python/tests/test_utils.py

```diff
@@ -38,304 +39,166 @@ def test_sampling_array_dims():
-    zone_a.name = 'ZONE-A'
-    zone_b = dlisio.plumbing.Zone()
-    zone_b.name = 'ZONE-B'
+@pytest.mark.parametrize('obj', [Parameter(), Computation()])
+def test_values_simple(obj):
+    obj.attic = {'VALUES' : [14], 'DIMENSION' : [1], 'ZONES' : []}
 
-    param = dlisio.plumbing.Parameter()
-    param.attic = {'VALUES' : [2], 'DIMENSION' : [1], 'AXIS' : [], 'ZONES' : []}
-
-    param.load()
-    param.zones = [zone_a, zone_b]
+    obj.load()
+    assert obj.values[0] == 14
 
-    assert param.values['DLISIO-UNDEF-0'] == 2
+@pytest.mark.parametrize('obj', [Parameter(), Computation()])
+def test_values_infer_simple(obj):
+    obj.attic = {'VALUES' : [14], 'DIMENSION' : [], 'ZONES' : []}
 
-def test_parameter_dimension1_value1():
-    param = dlisio.plumbing.Parameter()
-    param.attic = {'VALUES' : [14],
-                   'DIMENSION' : [1], 'AXIS' : [], 'ZONES' : []}
+    obj.load()
+    assert obj.values[0] == 14
 
-    param.load()
-    assert param.values['DLISIO-UNZONED'] == 14
+@pytest.mark.parametrize('obj', [Parameter(), Computation()])
+def test_values_one_sample(obj):
+    obj.attic = {'VALUES' : [1, 2], 'DIMENSION' : [2], 'ZONES' : []}
 
-def test_parameter_dimension_values():
-    param = dlisio.plumbing.Parameter()
-    param.attic = {'VALUES' : [1, 2],
-                   'DIMENSION' : [2], 'AXIS' : [], 'ZONES' : []}
+    obj.load()
+    assert list(obj.values[0]) == [1, 2]
 
-    param.load()
-    assert np.array_equal(param.values['DLISIO-UNZONED'], [1, 2])
+@pytest.mark.parametrize('obj', [Parameter(), Computation()])
+def test_values_wrong_dimensions(obj):
+    obj.attic = {'VALUES' : [1, 2, 3, 4, 5], 'DIMENSION' : [2], 'ZONES' : []}
 
-def test_parameter_wrong_dimension_values():
-    param = dlisio.plumbing.Parameter()
-    param.attic = {'VALUES' : [3],
-                   'DIMENSION' : [2], 'AXIS' : [], 'ZONES' : []}
+    obj.load()
+    msg  = 'cannot reshape array of size 5 into shape [2]'
+    with pytest.raises(ValueError) as error:
+        _ = obj.values
+    assert str(error.value) == msg
+    assert np.array_equal(obj.attic['VALUES'], np.array([1, 2, 3, 4, 5]))
 
-    param.load()
-    assert list(param.values['RAW']) == [3]
+@pytest.mark.parametrize('obj', [Parameter(), Computation()])
+def test_values_infer_dimensionsi_from_zones(obj):
+    obj.attic = {'VALUES' : [1, 2, 3, 4, 5], 'DIMENSION' : [], 'ZONES' : []}
 
-def test_parameter_wrong_dimension_wrong_zones_values():
-    zone_a = dlisio.plumbing.Zone()
-    zone_a.name = 'ZONE-A'
-    zone_b = dlisio.plumbing.Zone()
-    zone_b.name = 'ZONE-B'
+    obj.load()
 
-    param = dlisio.plumbing.Parameter()
-    param.attic = {'VALUES' : [1, 2, 3],
-                   'DIMENSION' : [1], 'AXIS' : [], 'ZONES' : []}
+    # Should be able to infer correct dimension from zones
+    obj.zones = [None for _ in range(5)]
+    assert np.array_equal(obj.values, np.array([1, 2, 3, 4, 5]))
+    assert np.array_equal(obj.attic['VALUES'], np.array([1, 2, 3, 4, 5]))
 
```

Comment:
Probably missing tests:
1. several values - yes, dimensions - no, zones - wrong.
2. one value - yes, dimensions - no, zones - wrong.
3. values - yes, dimensions - wrong, zones - wrong. In current implementation it's covered just by wrong dimensions test, but we might cover everything in case of future changes.
4. values - yes, dimensions - yes, zones - yes. (I don't see full happy path with zones present)